tests/integration/oidc/src/test/java/io/helidon/tests/integration/oidc/IntegrationTest.java

```diff
@@ -0,0 +1,133 @@
+package io.helidon.tests.integration.oidc;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.Response;
+
+import io.helidon.common.http.Http;
+import io.helidon.config.Config;
+import io.helidon.jersey.connector.HelidonProperties;
+import io.helidon.microprofile.server.Server;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@Testcontainers
+public class IntegrationTest {
+
+    private static final ClientConfig CONFIG = new ClientConfig().property(ClientProperties.FOLLOW_REDIRECTS, Boolean.TRUE)
+            .property(ClientProperties.CONNECT_TIMEOUT, 10000000)
+            .property(ClientProperties.READ_TIMEOUT, 10000000)
+            .property(HelidonProperties.CONFIG,
+                      Config.builder()
+                              .disableSystemPropertiesSource()
+                              .disableEnvironmentVariablesSource()
+                              .build()
+                              .get("client"));
+    private static final Client CLIENT = ClientBuilder.newClient(CONFIG);
+
+    @Container
+    public static KeycloakContainer keycloakContainer = new KeycloakContainer()
+            .withRealmImportFiles("/test-realm.json", "/test2-realm.json")
+            // this enables KeycloakContainer reuse across tests
+            .withReuse(true);
+
+    private static final int PORT = 7777;
+    private static Server server;
+
+    @BeforeAll
+    public static void beforeAll() {
+        System.setProperty("security.providers.1.oidc.identity-uri",
+                           keycloakContainer.getAuthServerUrl() + "realms/test/");
+        System.setProperty("security.providers.1.oidc.tenants.0.identity-uri",
+                           keycloakContainer.getAuthServerUrl() + "realms/test2/");
+        System.setProperty("security.providers.1.oidc.frontend-uri", "http://localhost:" + PORT);
+        server = Server.builder()
+                .port(PORT)
```

Comment:
Hardcoded port is not good for testing, is there a way to use a random port and set the properties post server start?